### PR TITLE
Use multi-line format in `@show`

### LINKED ISSUE #15820: Define macro @display
If you want to get a **readable** output of large arrays in Julia it's very convenient to use the `display()` function. You could use `show()`, but you won't be able to make much sense of a 1000 x 1000 element matrix  that way. There is a macro `@show`, but there is none for the `display()` function.

It would make sense in my opinion to add a macro that could look something like this:

```
macro display(val)
    display(val);println()
end
```

### PR DESCRIPTION
**Pull request by matthieugomez**
`@show` now display objects using the multi-line format.

This pull request fixes #15820 and this [Discourse issue](https://discourse.julialang.org/t/how-to-print-arrays-inside-functions/4091).  Replaces pull request #22244

### CODE REVIEW

File: test/show.jl [hunk 1]
```diff
@@ -764,3 +764,21 @@ end
 @test static_shown(Symbol("")) == "Symbol(\"\")"
 @test static_shown(Symbol("a/b")) == "Symbol(\"a/b\")"
 @test static_shown(Symbol("a-b")) == "Symbol(\"a-b\")"
+
+
+# Test @show
+fname = tempname()
+f = open(fname, "w")
```
Comment (tkelman) [inline test/show.jl]: would be good to use do-blocks for some of these, including the redirection so they get put back when finished

Comment (matthieugomez) [inline test/show.jl]: Ok I've pushed a new commit doing that. Is it why the tests did not pass?

Comment (tkelman) [inline test/show.jl]: it looks like this changes some output that test/test.jl is comparing against

**Comment by matthieugomez** [comment]
Tests are added and pass.

**Review by stevengj — APPROVED** [review]

**Comment by tkelman** [comment]
(should be squashed, on merge if not before)

**Comment by matthieugomez** [comment]
Thanks!

**Comment by StefanKarpinski** [comment]
I think we should consider reverting this. `@show` is particularly useful for debugging since you can often cut-and-paste the output to recreate the values but this change makes that far less common.

**Comment by iamed2** [comment]
I don't like how this turned out, for two reasons:

1. `@show x` now uses a different method than `show(x)` to display the result
2. `@show x` was helpful for doctesting for loops in IterTools.jl, but the doctest output would now be so ugly and long that we'll have to change it

I propose bringing back https://github.com/JuliaLang/julia/pull/22244 (`@display` macro)

Julia 0.6:
```julia
julia> show(x)
[1, 2, 3]
julia> show(STDOUT, "text/plain", x)
3-element Array{Int64,1}:
 1
 2
 3
julia> @show x;
x = [1, 2, 3]
```
Julia 0.7:
```julia
julia> show(x)
[1, 2, 3]
julia> show(STDOUT, "text/plain", x)
3-element Array{Int64,1}:
 1
 2
 3
julia> @show x;
x = 3-element Array{Int64,1}:
 1
 2
 3
```

**Comment by StefanKarpinski** [comment]
Triage agrees: let's revert this.

**Pull request merged**